src/shared/extensionUtilities.ts

```diff
@@ -123,3 +125,58 @@ function convertExtensionRootTokensToPath(
 ): string {
     return text.replace(/!!EXTENSIONROOT!!/g, `vscode-resource:${basePath}`)
 }
+
+/**
+ * Utility function to determine if the extension version has changed between activations
+ * False (versions are identical) if version key exists in global state, is a semver, and matches the current version
+ * True (versions are different) if any of the above aren't true
+ *
+ * TODO: Change the threshold on which we display the welcome page?
+ * For instance, if we start building nightlies, only show page for significant updates?
+ *
+ * @param context VS Code Extension Context
+ * @param currVersion Current version to compare stored most recent version against (useful for tests)
+ */
+export function isDifferentVersion(context: vscode.ExtensionContext, currVersion: string = pluginVersion): boolean {
+    const mostRecentVersion = context.globalState.get<string>(mostRecentVersionKey)
+    if (mostRecentVersion && semver.valid(mostRecentVersion) && mostRecentVersion === currVersion) {
+        return false
+    }
+
+    return true
+}
+
+/**
+ * Utility function to update the most recently used extension version
+ * Pulls from package.json
+ *
+ * @param context VS Code Extension Context
+ */
+export function setMostRecentVersion(context: vscode.ExtensionContext): void {
+    context.globalState.update(mostRecentVersionKey, pluginVersion)
+}
+
+/**
+ * Publishes a toast with a link to the welcome page
+ */
+export async function promptWelcome(): Promise<void> {
+    const view = localize(
+        'AWS.message.prompt.welcome.openPage',
+        'View Welcome Page'
+    )
+    const prompt = await vscode.window.showInformationMessage(
+        localize(
+            'AWS.message.prompt.welcome.toastMessage',
+            'You are now using the AWS Toolkit for Visual Studio Code, version {0}',
+            pluginVersion
+        ),
+        view,
+        localize(
```

Comment:
The dismiss button seems superfluous. There is an 'X' on the toast in the top right (pretty much above this button), which is standard VS Code UX.

Here is an example of a dialog you'd similarly dismiss if you didn't want to perform the suggested action:
![image](https://user-images.githubusercontent.com/39839589/59790518-f24d4780-9284-11e9-8956-093839b8b1ee.png)